src/Destructurama.Attributed/Attributed/NotLoggedIfDefaultAttribute.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright 2020 Destructurama Contributors, Serilog Contributors
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Serilog.Core;
+using Serilog.Events;
+#if NETSTANDARD1_1
+using System.Reflection;
+#endif
+
+namespace Destructurama.Attributed
+{
+    /// <summary>
+    /// Specified that a property should not be included when destructuring an object for logging.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property)]
+    public class NotLoggedIfDefaultAttribute : Attribute, IPropertyDestructuringAttribute
+    {
+        public bool TryCreateLogEventProperty(string name, object value, ILogEventPropertyValueFactory propertyValueFactory, out LogEventProperty property)
+        {
+            if (value != null)
+            {
+                var type = value.GetType();
+                object typeDefaultValue = null;
+
+#if NETSTANDARD1_1
+                if (type.GetTypeInfo().IsValueType)
+#else
+                if (type.IsValueType)
+#endif
+                    typeDefaultValue = Activator.CreateInstance(type);
```

Comment:
Costly implementation - memory allocation/boxing for each call.

Reply:
I will try to find alternative solutions to compare with.
@sungam3r do you have any suggestions? 
I will appreciate it if you share your thoughts!

Reply:
In fact, there is a more serious problem, by solving which this code of possibility will be changed.
You use type of **value** to determine _default_ value. In case of `public object A { get; set; } = 0;` it is wrong. You do need information about **property** type.

Reply:
I have changed the implementation a little bit!
Introduced a Cache to construct objects once.
@sungam3r Tell me what you think about it!

Reply:
I think that you introduce a breaking change into public API. I don’t know if such changes will be accepted.

Reply:
Ok I can introduce another interface that will include type and thus it will not be a breaking change!
What's your opinion @nblumhardt ?

Reply:
Hello, was this ever merged into the main branch? I would love to be able to ignore null/empty properties. Thanks